x/concentrated-liquidity/lp.go

```diff
@@ -428,12 +428,11 @@ func (k Keeper) isLockMature(ctx sdk.Context, underlyingLockId uint64) (bool, er
 		return false, err
 	}
 
-	// Check if the lock has expired
-	if underlyingLock.EndTime.After(ctx.BlockTime()) {
-		// Lock is still active, so we cannot withdraw from this position
+	if underlyingLock.EndTime.IsZero() {
```

Comment:
Is `EndTime` not set when the lock is active?

Reply:
Correct, the way you can verify a lock is active is the end time is the zero value for time.Time